### PR TITLE
Value hints

### PR DESCRIPTION
**Pull request by MarcelGarus**
Depends on #228

### CODE REVIEW

File: compiler/src/language_server/mod.rs [hunk 1]
```diff
@@ -25,25 +31,29 @@ use lsp_types::{
...
+            db: Default::default(),
+            hints_server_sink: Default::default(),
+        }
+    }
+
+    async fn send_to_hints_server(&self, event: hints::Event) {
+        let event_sink = self.hints_server_sink.lock().await;
+        match event_sink.as_ref().unwrap().send(event).await {
+            Ok(_) => {}
+            Err(_) => panic!("Couldn't shutdown hints server."),
```
Comment (JonasWanke) [inline compiler/src/language_server/mod.rs]:
```suggestion
            Err(_) => panic!("Couldn't send message to hints server."),
```

File: packages/Test.candy.hints [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Hint { kind: Value, text: " # [ 0: 110, 10: 97, 16: 50, 20: 111, 24: 44, 4: 44, 6: 103, 3: 101, 7: 101, 13: 101, 22: 97, 21: 110, 9: 77, 23: 115, 17: 50, 14: 108, 19: 74, 5: 97, 8: 10, 11: 114, 25: 50, 12: 99, 26: 50, 18: 10, 1: 97, 15: 44, 2: 109 ]", position: Position { line: 0, character: 28 } }
```
Comment (JonasWanke) [inline packages/Test.candy.hints:1]: I think we can `.gitignore` the generated `*.candy.*` files

File: compiler/src/vm/use_provider.rs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            .map(|bytes| (*bytes).clone())
+            .ok_or_else(|| format!("Couldn't import file '{}'.", input))
+    }
+
+    fn use_local_module(&self, input: Input) -> Option<Lir> {
+        self.db.lir(input).map(|lir| (*lir).clone())
+    }
+}
+
+pub struct FunctionUseProvider<'a, U1, U2>
```
Comment (JonasWanke) [inline compiler/src/vm/use_provider.rs]: This does not appear to be used anywhere. Do we need it or was it just temporary?

**Review by JonasWanke — CHANGES_REQUESTED** [review]

File: packages/Test.candy.hints (hunk 2 above)
Comment (MarcelGarus) [reply packages/Test.candy.hints:1]: Oh, definitely, good idea. I have several of these files which I always consciously ignore in GitHub Desktop, but seems like one slipped through.

File: compiler/src/vm/builtin_functions.rs [hunk 4]
```diff
@@ -135,7 +135,7 @@ impl Vm {
 
     fn print(&mut self, args: Vec<Value>) -> Result<Value, String> {
         destructure!(args, [Value::Text(message)], {
-            println!("{message:?}");
+            log::info!("{message:?}");
             Ok(Value::nothing())
         })
```
Comment (JonasWanke) [inline compiler/src/vm/builtin_functions.rs:140]: I think this broke, maybe during a merge?

**Review by JonasWanke — CHANGES_REQUESTED** [review]

File: compiler/src/vm/builtin_functions.rs (hunk 4 above)
Comment (MarcelGarus) [reply compiler/src/vm/builtin_functions.rs:140]: What's the problem here? This seems fine to me

Comment (JonasWanke) [reply compiler/src/vm/builtin_functions.rs:140]: Interesting, this was just showing me a different diff… Then it LGTM

**Review by JonasWanke — APPROVED** [review]

**Pull request merged**